lib/asset_cloud/base.rb

```diff
@@ -8,8 +8,7 @@ class IllegalPath < StandardError
 
   class Base
     cattr_accessor :logger
-
-    VALID_PATHS = /\A[a-z0-9][a-z0-9_\-\/]+([a-z0-9][\w\-\ \.\@]*\.\w{2,6})?\z/i
+    VALID_PATHS = /\A(([a-z0-9])|((\.?[\w\-\@])([\w\-\@]|[\ ][\w\-\@]|[\/][\w\-\@\.]|[\.][\w\-\@]+)*))\z/i
```

Comment:
It's actually not ugly as it initially appears... It's effectively a case statement written in regex form.

  (
    ([a-z0-9]) #One Letter Filenames
    |
    (
      (\.?[\w\-\@]) #It can start with a dot but it must have a following character
      (
        [\w\-\@] #You can have a letter
        |
        [\ ][\w\-\@] #If there is a space you need to have a normal letter afterward
        |
        [\/][\w\-\@] #If there is a slash you need to have a normal letter afterward
        |
        [\/][\.][\w\-\@] #Though a slash could be followed by a dot so long as there is a normal letter afterward
        |
        [\.][\w\-\@]+ #A dot must be followed by one (or more) normal letters
      )* #Zero to many of these combinations.
    )
  )

Actually maybe I'll put it in the source broken up and commented like that.